File: Núcleo/src/LogicaNegocio/Usuario.java
package LogicaNegocio;

import Recursos.DatosUsuario;

/**
 * @author Victor Javier
 * @version 1.0
 * @created 18-may.-2019 06:27:03 p. m.
 */
public class Usuario {

    private int idUsuario;
    private String nombreUsuario;
    private String contraseña;
    private Personal personalUsuario;
    private IUsuario iUsuario;
    
    private DatosUsuario validarDatos(){
        DatosUsuario datosUsuario = DatosUsuario.VALIDO;
        if (this.idUsuario < 1){
            datosUsuario = DatosUsuario.ID_NEGATIVO;
        } else if (this.nombreUsuario.trim().equals("") || this.nombreUsuario == null){
            datosUsuario = DatosUsuario.NOMBRE_VACIO;
        } else if (this.nombreUsuario.trim().length() > 15){
            datosUsuario = DatosUsuario.NOMBRE_LARGO;
        } else if (this.contraseña.trim().equals("") || this.contraseña == null){
            datosUsuario = DatosUsuario.CONTRASEÑA_VACIA;
        } else if (this.contraseña.trim().length() < 6){
            datosUsuario = DatosUsuario.CONTRASEÑA_CORTA;
        } else if (this.contraseña.trim().length() > 15){
            datosUsuario = DatosUsuario.CONTRASEÑA_LARGA;
        }
        return datosUsuario;
    }

    public Usuario() {
        this.idUsuario = 0;
        this.nombreUsuario = "";
        this.contraseña = "";
        this.iUsuario = null;
    }

    public int getIdUsuario() {
        return idUsuario;
    }

    public void setIdUsuario(int idUsuario) {
        this.idUsuario = idUsuario;
    }

    public String getNombreUsuario() {
        return nombreUsuario;
    }

    public void setNombreUsuario(String nombreUsuario) {
        this.nombreUsuario = nombreUsuario;
    }

    public String getContraseña() {
        return contraseña;
    }

    public void setContraseña(String contraseña) {
        this.contraseña = contraseña;
    }

    public Personal getPersonalUsuario() {
        return personalUsuario;
    }

    public void setPersonalUsuario(Personal personalUsuario) {
        this.personalUsuario = personalUsuario;
    }

    public IUsuario getiUsuario() {
        return iUsuario;
    }

    public void setiUsuario(IUsuario iUsuario) {
        this.iUsuario = iUsuario;
    }
    
    public DatosUsuario registrarUsuario() {
        DatosUsuario validacion = this.validarDatos();
        if (validacion == DatosUsuario.VALIDO){
            validacion = this.iUsuario.registrarUsuario(this) ? DatosUsuario.EXITO : DatosUsuario.ERROR_ALMACENAMIENTO;
        }
        return validacion;
    }

    public DatosUsuario modificar() {
        DatosUsuario modificacion = this.validarDatos();
        if (modificacion == DatosUsuario.VALIDO){
            modificacion = this.iUsuario.modificar(this) ? DatosUsuario.EXITO : DatosUsuario.ERROR_ALMACENAMIENTO;
        }
        return modificacion;
    }

    /**
     *
     * @param nombre
     * @param contraseña
     * @return 
     */
    public int iniciarSesion(String nombre, String contraseña) {
        return this.iUsuario.iniciarSesion(nombre, contraseña);
    }
}//end Usuario

File: Nucleo servidor web/application/libraries/Usuario.php
<?php
class Usuario{
    private $idUsuario;
    private $nombreUsuario;
    private $contraseña;
    private $personalUsuario;
    private $iUsuario;
    private $rol;

    public function __get($attr) {
		return CI_Controller::get_instance()->$attr;
    }
    
    public function __construct(){
        
    }

    private function validarDatos(){
        $datosUsuario = DatosUsuario::VALIDO;
        if ($this->idUsuario < 1){
            $datosUsuario = DatosUsuario::ID_NEGATIVO;
        } else if ($this->nombreUsuario == "" || $this->nombreUsuario == null){
            $datosUsuario = DatosUsuario::NOMBRE_VACIO;
        } else if (strlen($this->nombreUsuario) > 15){
            $datosUsuario = DatosUsuario::NOMBRE_LARGO;
        } else if ($this->contraseña == "" || $this->contraseña == null){
            $datosUsuario = DatosUsuario::CONTRASEÑA_VACIA;
        } else if (strlen($this->contraseña) < 6){
            $datosUsuario = DatosUsuario::CONTRASEÑA_CORTA;
        } else if (strlen($this->contraseña) > 15){
            $datosUsuario = DatosUsuario::CONTRASEÑA_LARGA;
        }
        return $datosUsuario;
    }
    public function getIdUsuario() {
        return $this->idUsuario;
    }

    public function setIdUsuario($idUsuario) {
        $this->idUsuario = $idUsuario;
    }

    public function getNombreUsuario() {
        return $this->nombreUsuario;
    }

    public function setNombreUsuario($nombreUsuario) {
        $this->nombreUsuario = $nombreUsuario;
    }

    public function getContraseña() {
        return $this->contraseña;
    }

    public function setContraseña($contraseña) {
        $this->contraseña = $contraseña;
    }

    public function getPersonalUsuario() {
        return $this->personalUsuario;
    }

    public function setPersonalUsuario($personalUsuario) {
        $this->personalUsuario = $personalUsuario;
    }

    public function getiUsuario() {
        return $this->iUsuario;
    }

    public function setiUsuario($iUsuario) {
        $this->iUsuario = $iUsuario;
    }
    public function getRol(){
        return $this->rol;
    }
    public function setRol($rol){
        $this->rol = $rol;
    }
    
    public function registrarUsuario() {
        $validacion = $this->validarDatos();
        if ($validacion == DatosUsuario::VALIDO){
            $validacion = $this->iUsuario->registrarUsuario($this) ? DatosUsuario::EXITO : DatosUsuario::ERROR_ALMACENAMIENTO;
        }
        return $validacion;
    }

    public function modificar() {
        $modificacion = $this->validarDatos();
        if ($modificacion == DatosUsuario::VALIDO){
            $modificacion = $this->iUsuario->modificar($this) ? DatosUsuario::EXITO : DatosUsuario::ERROR_ALMACENAMIENTO;
        }
        return $modificacion;
    }

    /**
     *
     * @param nombre
     * @param contraseña
     * @return 
     */
    public function iniciarSesion() {
        return $this->iUsuario->iniciarSesion($this->getNombreUsuario(), $this->getContraseña());
    }

}